File: backend/app/schemas/health.py
"""Health-check schemas.

HealthCheckOut is the read shape returned by the list and manual-trigger
routes. There is no create-input schema: checks are produced by the backend
(domain.health_checker), never supplied by the client.
"""
import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict

from app.models.health import HealthCheckStatus


class HealthCheckOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    service_id: uuid.UUID
    status_code: int | None
    response_time_ms: int | None
    status: HealthCheckStatus
    error_message: str | None
    checked_at: datetime
    created_at: datetime
    updated_at: datetime

File: frontend/src/types/health.ts
export type HealthCheckStatus = "healthy" | "degraded" | "down";

export interface HealthCheck {
  id: string;
  service_id: string;
  status_code: number | null;
  response_time_ms: number | null;
  status: HealthCheckStatus;
  error_message: string | null;
  checked_at: string;
  created_at: string;
  updated_at: string;
}
